File: src/main/java/me/nobaboy/nobaaddons/commands/subcommands/SSAverageCommand.java
package me.nobaboy.nobaaddons.commands.subcommands;

import me.nobaboy.nobaaddons.commands.ISubCommand;
import me.nobaboy.nobaaddons.features.dungeons.data.SSFile;
import me.nobaboy.nobaaddons.util.ChatUtils;

import java.util.Collections;
import java.util.List;

public class SSAverageCommand implements ISubCommand {
	@Override
	public String getName() {
		return "ssAverage";
	}

	@Override
	public List<String> getAliases() {
		return Collections.singletonList("ssAvg");
	}

	@Override
	public void run(String[] args) {
		double sum = 0;
		long count = 0;

		List<Double> times = SSFile.INSTANCE.times.get();
		for(Double time : times) {
			sum += time;
			count++;
		}
		if(count == 0) {
			ChatUtils.addMessage("You have not completed a single Simon Says device in the Catacombs Floor 7.");
			return;
		}
		double average = Math.round((sum / count) * 1000.0) / 1000.0;
		ChatUtils.addMessage("Your average time for Simon Says is: " + average + "s (Total SS Devices: " + count + ")");
	}
}

File: src/main/kotlin/me/nobaboy/nobaaddons/commands/subcommands/SSAverageCommand.kt
package me.nobaboy.nobaaddons.commands.subcommands

import me.nobaboy.nobaaddons.commands.ISubCommand
import me.nobaboy.nobaaddons.features.dungeons.data.SSFile
import me.nobaboy.nobaaddons.util.ChatUtils

class SSAverageCommand : ISubCommand {
    override val name: String = "ssAverage"

    override val aliases: List<String> = listOf("ssAvg")

    override fun run(args: Array<String>) {
        val times = SSFile.times
        val count = times.size

        if (count == 0) {
            ChatUtils.addMessage("You have not completed a single Simon Says device in the Catacombs Floor 7.")
            return
        }

        val sum = times.sum()
        val average = sum / count

        val formattedAverage = "%.3f".format(average)
        ChatUtils.addMessage("Your average time for Simon Says is: ${formattedAverage}s (Total SS Devices: $count)")
    }
}